pallets/messages/src/mock.rs

```diff
@@ -118,6 +119,14 @@ impl AccountProvider for AccountHandler {
 		let info = KeyInfo { msa_id: get_msa_from_account(*key), nonce: 0, expired: 100 };
 		Ok(info)
 	}
+
+	fn ensure_valid_delegation(provider: Provider, _delegator: Delegator) -> DispatchResult {
+		if provider == Provider(2000) {
+			return Err(DispatchError::Other("some delegation error"))
+		};
+
```

Comment:
I think it is ok but message pallet will need permissions data which was returned previously to make decision around a message is allowed to be published or not given the permissions/grants
So this will require refactoring again